apps/web/src/pages/Applications/ApplicationSkillsPage/components/SkillForm.tsx

```diff
@@ -0,0 +1,316 @@
+};
+
+type FormAction = "connect" | "remove";
+
+type FormValues = {
+  experience?: Scalars["ID"];
+  skill?: Scalars["ID"];
+  details?: string;
+  action?: FormAction;
+};
+
+interface SkillFormProps {
+  defaultValues: FormValues;
+  experiences: Experience[];
+  onSuccess: () => void;
+}
+
+const SkillForm = ({
+  defaultValues,
+  onSuccess,
+  experiences,
+}: SkillFormProps) => {
+  const intl = useIntl();
+  const methods = useForm<FormValues>({
+    defaultValues,
+  });
+  const { register, setValue } = methods;
+  const actionProps = register("action");
+  const selectedExperienceId = methods.watch("experience");
+  const selectedExperience = experiences.find(
+    (exp) => exp.id === selectedExperienceId,
+  );
+  const experienceType = selectedExperience
+    ? deriveExperienceType(selectedExperience)
+    : "";
+  const { executeMutation, getMutationArgs } = useExperienceMutations(
+    "update",
+    experienceType,
+  );
+
+  const handleSubmit = (formValues: FormValues) => {
+    const args = getMutationArgs(
+      formValues.experience || "",
+      formValues.skill
+        ? {
+            skills: getSkillArgs(
+              formValues.skill,
+              selectedExperience,
+              formValues.details,
+              formValues.action === "remove",
+            ),
+          }
+        : {},
+    );
+    if (executeMutation) {
+      executeMutation(args)
+        .then((res) => {
+          if (res.data) {
+            toast.success(
+              intl.formatMessage({
+                defaultMessage: "Successfully linked experience!",
+                id: "SEdLz1",
+                description:
+                  "Success message displayed after linking an experience to a skill",
+              }),
+            );
+            onSuccess();
+          }
+        })
+        .catch(() => {
+          toast.error(
+            intl.formatMessage({
+              defaultMessage: "Error: linking experience failed",
+              id: "a3epXH",
+              description:
+                "Message displayed to user after experience fails to be linked to a skill.",
+            }),
+          );
+        });
+    }
+  };
+
+  return (
+    <FormProvider {...methods}>
+      <form onSubmit={methods.handleSubmit(handleSubmit)}>
+        <input type="hidden" {...methods.register("skill")} />
+        <Heading level="h2" size="h6" data-h2-margin-top="base(0)">
+          {intl.formatMessage({
+            defaultMessage: "Choose the experience you'd like to add",
+            id: "dkYHTv",
+            description: "Heading for the experience selection",
+          })}
+        </Heading>
+        <Select
+          id="experience"
+          name="experience"
+          disabled={!!defaultValues.experience}
+          label={intl.formatMessage({
+            defaultMessage: "Select a type of experience to add",
+            id: "plC0m5",
+            description: "Label for the experience select input",
+          })}
```

Comment:
should be, from Figma

https://user-images.githubusercontent.com/4313717/227238454-c93863a4-12fa-4693-827e-f4a533a8d7db.png

Reply:
fixed in [fix typos](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6518/commits/edc26071cb1e65b4b1fd46f6deab189f541d0c61)